### PR TITLE
fix go type name for list

### PR DESCRIPTION
**Pull request by wongoo**
**What this PR does**:
- fix go type name for list

**Does this PR introduce a user-facing change?**:

```release-note
- fix go type name for list
```

### CODE REVIEW

File: pojo_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hessian
+
+import (
```
Comment (AlexStocks) [inline pojo_test.go:20]: haha, split it, pls.

File: pojo_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	"testing"
+	"time"
+)
+
+import (
+	"github.com/apache/dubbo-go-hessian2/java_exception"
+	"github.com/apache/dubbo-go-hessian2/java_util"
+)
+
+import (
```
Comment (AlexStocks) [inline pojo_test.go:32]: pls move this block up as the 2rd import block.

**Review by justxuewei — APPROVED** [review]

File: pojo.go [hunk 3]
```diff
@@ -244,20 +240,28 @@ func unRegisterPOJO(o POJO) int {
...
-	for reflect.Ptr == goType.Kind() {
-		goType = goType.Elem()
-	}
-	return combineGoName(goType)
+// GetGoName get the raw go type name with package.
+func GetGoName(o interface{}) string {
+	return combineGoName(reflect.TypeOf(o))
 }
 
 func combineGoName(t reflect.Type) string {
```
Comment (AlexStocks) [inline pojo.go:248]: combineGoTypeName

File: pojo.go [hunk 4]
```diff
@@ -244,20 +240,28 @@ func unRegisterPOJO(o POJO) int {
...
 }
 
-func getGoName(o interface{}) string {
-	goType := reflect.TypeOf(o)
-	for reflect.Ptr == goType.Kind() {
-		goType = goType.Elem()
-	}
-	return combineGoName(goType)
+// GetGoName get the raw go type name with package.
+func GetGoName(o interface{}) string {
```
Comment (AlexStocks) [inline pojo.go:244]: the func name GetGoType maybe more perfect.

**Review by AlexStocks — APPROVED** [review]

**Review by zouyx — APPROVED** [review]

**Pull request merged**